pkg/cri/server/sandbox_run_linux.go

```diff
@@ -101,12 +101,13 @@ func (c *criService) sandboxContainerSpec(id string, config *runtime.PodSandboxC
 	if nsOptions.GetIpc() == runtime.NamespaceMode_NODE {
 		sandboxDevShm = devShm
 	}
+	specOpts = append(specOpts, oci.WithoutMounts("/dev/shm"))
 	specOpts = append(specOpts, oci.WithMounts([]runtimespec.Mount{
 		{
 			Source:      sandboxDevShm,
 			Destination: devShm,
 			Type:        "bind",
-			Options:     []string{"rbind", "ro"},
+			Options:     []string{"rbind", "rw"},
```

Comment:
For writing what?

Reply:
If you do not intend to write anything, you may just need to append these flags

https://github.com/moby/moby/blob/v20.10.5/daemon/oci_linux.go#L420-L450

Reply:
Thanks @AkihiroSuda , with help from you I found the root cause of the failure.
In linux/fs/namespace.c:do_remount(), it checks the remount operation doesn't reset (instead of modifying) RO,NOSUID,NODEV,NOEXEC and MNT_ATIME_xxx. 
```
        /* Don't allow changing of locked mnt flags.
         *
         * No locks need to be held here while testing the various
         * MNT_LOCK flags because those flags can never be cleared
         * once they are set.
         */
        if ((mnt->mnt.mnt_flags & MNT_LOCK_READONLY) &&
            !(mnt_flags & MNT_READONLY)) {
                return -EPERM;
        }
        if ((mnt->mnt.mnt_flags & MNT_LOCK_NODEV) &&
            !(mnt_flags & MNT_NODEV)) {
                return -EPERM;
        }
        if ((mnt->mnt.mnt_flags & MNT_LOCK_NOSUID) &&
            !(mnt_flags & MNT_NOSUID)) {
                return -EPERM;
        }
        if ((mnt->mnt.mnt_flags & MNT_LOCK_NOEXEC) &&
            !(mnt_flags & MNT_NOEXEC)) {
                return -EPERM;
        }
        if ((mnt->mnt.mnt_flags & MNT_LOCK_ATIME) &&
            ((mnt->mnt.mnt_flags & MNT_ATIME_MASK) != (mnt_flags & MNT_ATIME_MASK))) {
                return -EPERM;
        }
```

Originally the `/dev/shm` was mount as 
```shm on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)```.

So when doing in bind mounting in user namespace, in addition to changing `rw` to `ro`, we still need those "nosuid,nodev,noexec,relatime" flags. With those flags, it succeeds:
```
root@f8ff64bfc7d8:/kata# mount -o bind,ro,nosuid,nodev,noexec,relatime /dev/shm /dev/shm
root@f8ff64bfc7d8:/kata# mount -o bind,ro /dev/shm /dev/shm
mount: /dev/shm: filesystem was mounted, but any subsequent operation failed: Unknown error 5005.
```

So we have a better solution now and will update the patch.

Reply:
One more information for the root cause. When the mount flags contains "BIND | READONLY", it will cause a second mount syscall with "mount_flags | REMOUNT", which will handled by `do_remount()`, though causes the failure. By changing "ro" to "rw", the second mount syscall will not happen, so it succeeds.